Data/sync/Sync.swift

```diff
@@ -448,52 +448,46 @@ extension Sync {
             fetchedRecords = fetchedRecords.sorted { device1, device2 in
                 device1.syncTimestamp ?? -1 > device2.syncTimestamp ?? -1 }.unique { $0.objectId ?? [] == $1.objectId ?? [] }
             
-        } else if recordType == .bookmark {
-            // Bookmarks are sorted to have the oldest bookmark to come in first, so it can be added before it's children.
-            fetchedRecords = fetchedRecords.sorted(by: {
-                let firstTimestamp = $0.syncTimestamp ?? 0
-                let secondTimestamp = $1.syncTimestamp ?? 0
-                
-                return firstTimestamp < secondTimestamp
-            })
         }
         
         let context = DataController.newBackgroundContext()
-        for fetchedRoot in fetchedRecords {
-            
-            guard
-                let fetchedId = fetchedRoot.objectId
-                else { return }
-            
-            let clientRecord = recordType.coredataModelType?.get(syncUUIDs: [fetchedId], context: context)?.first as? Syncable
-            
-            var action = SyncActions(rawValue: fetchedRoot.action ?? -1)
-            if action == SyncActions.delete {
-                clientRecord?.remove(sendToSync: false)
-            } else if action == SyncActions.create {
+        context.perform {
+            for fetchedRoot in fetchedRecords {
```

Comment:
Just to confirm, I do not see any logic changes inside this (just that it is wrapped inside a `.perform` is this a correct observation?

Reply:
Yes sir.